parser/structured_data.go

```diff
@@ -51,53 +51,46 @@ func (sd *StructuredData) Parse(bytes []byte) (*skogul.Container, error) {
 	}, nil
 }
 
-// mnrParseData takes the raw input and parses it
+// parseStructuredData takes the raw input and parses it
 // this takes care of splitting input on newlines etc
 func (sd *StructuredData) parseStructuredData(data []byte) ([]*skogul.Metric, error) {
-	lines := bytes.Split(data, []byte("\n"))
+	lineScanner := bufio.NewScanner(bytes.NewReader(data))
+	lineScanner.Split(splitStructuredDataMetrics)
 
 	metrics := make([]*skogul.Metric, 0)
 
 	timestamp := skogul.Now()
-
-	for _, l := range lines {
-		line := bytes.TrimSpace(l)
+	for lineScanner.Scan() {
+		line := lineScanner.Bytes()
 		if len(line) == 0 {
 			// Skip empty lines
 			continue
 		}
 
+		metric := &skogul.Metric{
+			Time:     &timestamp,
+			Metadata: make(map[string]interface{}),
+			Data:     make(map[string]interface{}),
+		}
+
+		// Set up the Key-Value scanner to extract data
 		kvScanner := bufio.NewScanner(bytes.NewReader(line))
 		kvScanner.Split(splitKeyValuePairs)
 
-		var metric *skogul.Metric
-		for {
-			canContinue := kvScanner.Scan()
-
+		for kvScanner.Scan() {
 			tag := strings.Trim(kvScanner.Text(), "\u0000")
 			tagValue := strings.SplitN(tag, "=", 2)
 
+			if len(tagValue) == 1 && metric.Metadata["sd-id"] == nil {
```

Comment:
wat?

I take it this is some magic structured data detail, but what's sd-id and what's going on here? Some comment, and possibly user-documentation seems reasonable.

Reply:
I can refer to the RFC for what the SD-ID is, but this line of code was just moved from here:
https://github.com/telenornms/skogul/commit/35dce7c3ba03e82ad6e4fed0365dd53425d47608#diff-d88400ccea9a24abee535154bbe87f6e872ac0555ca0e765f8315c39c929b1d1L99

Reply:
Since parsers don't have configurable fields it's a bit hard to do any more than that... 🤔

Reply:
I didn't spot the move, that's on me.

However: Parser _can_ have configurable fields now like all modules, just, this one doesn't. You could, however, mention it in parser/auto.go .  But go ahead and merge if you feel like it, since this isn't a part of the PR.